distributor/distributor.go

```diff
@@ -173,12 +173,21 @@ type sampleTracker struct {
 
 // Push implements cortex.IngesterServer
 func (d *Distributor) Push(ctx context.Context, req *remote.WriteRequest) (*cortex.WriteResponse, error) {
-	userID, err := user.GetID(ctx)
+	samples := util.FromWriteRequest(req)
+	err := d.AppendMany(ctx, samples)
 	if err != nil {
 		return nil, err
 	}
+	return &cortex.WriteResponse{}, nil
+}
+
+// AppendMany appends many samples to the given context.
+func (d *Distributor) AppendMany(ctx context.Context, samples []*model.Sample) error {
```

Comment:
The use of model.Sample will be going away from the distributor, I intend for the distributor to only work with the wire format - theres no point in it converting from/to prometheus' internal format.